autoscaler/controllers/gateway/config/root.go

```diff
@@ -38,7 +39,14 @@ func Calculate(dests *odigosv1.DestinationList, processors *odigosv1.ProcessorLi
 			return "", fmt.Errorf("no configer for %s", dest.Spec.Type)
 		}
 
-		configer.ModifyConfig(&dest, currentConfig)
+		err := configer.ModifyConfig(&dest, currentConfig)
+		if err != nil {
+			log.Log.V(0).Error(err, "Error:", err)
+			dest.Status = err.Error() // Update the status with the error message
```

Comment:
`dest.Status` is of type:
```go
type DestinationStatus struct {
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
}
```

So I assume this isn't compiling.

You can catch a lot of issues even before review by compiling the code and checking errors from the compiler.

To populate the Status conditions, you can see how it is done [here](https://github.com/keyval-dev/odigos/blob/37987d1f9849785519592d66ab7ab70a1dfeff65/autoscaler/controllers/actions/addclusterinfo_controller.go#L79)